### PR TITLE
Improved manual test in `HtmlComment` plugin and added to both all-features tests

### LINKED ISSUE #10154: HTML Comments: Improve the manual test and add it to both all-features tests
We need some scenarios and more complex initial content

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (html-support): Improved manual test in `HtmlComment` plugin and added plugin to both all-features tests. Closes #10154.

### CODE REVIEW

File: packages/ckeditor5-html-support/tests/manual/htmlcomment.md [hunk 1]
```diff
@@ -1 +1,13 @@
 ## HTML comment
+
+1. Toggle the source editing mode and verify that all comments are present. There should be 10 comments.
+1. Open CKEditor 5 inspector and verify, that there are 10 $root attributes, one for each comment. The value set for each attribute is the comment's content (from `C1` to `C10`).
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/tests/manual/htmlcomment.md:4]: We should focus here on the functionalities, not the implementation. This test is rather unnecessary.

File: packages/ckeditor5-html-support/tests/manual/htmlcomment.md [hunk 2]
```diff
@@ -1 +1,13 @@
...
+1. In source editing mode add and remove some comments. Please note that, currently, the comment support is on a basic level. See the **Known limitations** section below containing missing functionalities.
+1. Copy & paste some content from Word and Google Docs. Content should be parsed without errors and displayed correctly.
+
+### Known limitations
+
+1. [Issue #10118](https://github.com/ckeditor/ckeditor5/issues/10118): comments located between some HTML tags are repositioned or lost.
+1. [Issue #10119](https://github.com/ckeditor/ckeditor5/issues/10119): comments can be easily repositioned during editing the document.
+1. [Issue #10152](https://github.com/ckeditor/ckeditor5/issues/10152): `editor.getData()` cuts out all comments when there is no more content.
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/tests/manual/htmlcomment.md:12]: Let's remove this one.

File: tests/manual/all-features.md [hunk 3]
```diff
@@ -42,6 +42,8 @@ The "previews in view" mode is enabled, which means that previews should be visi
 
 **Text part language** - There are 2 paragraphs with applied language: Spanish with left-to-right direction and Arabic with right-to-left direction. Both should have italic style. Play with the selection and the language dropdown.
 
+**HTML comments** - There are 10 comments in this section: from `C1` to `C10`. Open the source editing mode (or just call the `editor.getData()`) and verify that they are present. Open CKEditor 5 inspector and verify, that there are 10 $root attributes, one for each comment.
```
Comment (ma2ciek) [inline tests/manual/all-features.md:45]: Same here, I'd remove the section about root attributes.

File: packages/ckeditor5-html-support/tests/manual/htmlcomment.md (hunk 1 above)
Comment (psmyrek) [reply packages/ckeditor5-html-support/tests/manual/htmlcomment.md]: Done.

File: packages/ckeditor5-html-support/tests/manual/htmlcomment.md (hunk 2 above)
Comment (psmyrek) [reply packages/ckeditor5-html-support/tests/manual/htmlcomment.md]: Done.

File: tests/manual/all-features.md (hunk 3 above)
Comment (psmyrek) [reply tests/manual/all-features.md]: Done.

**Comment by ma2ciek** [comment]
LGTM.

**Pull request merged**